### ISSUE TITLE
Reconsider how we refer to X and y in DataOps

### ISSUE THREAD

**Issue by rcap107** [skrub-data/skrub#1960]
The docstrings for mark_as_X and mark_as_y refer to X and y as "tables", but Data Ops do not require to be used only with tables. We should try to use a more "data-agnostic" term, so that we can also refer to images for example. 

> maybe this should not be called a table, since data ops can be used with non-tabular data like images? 

 _Originally posted by @rcap107 in [#1943](https://github.com/skrub-data/skrub/pull/1943/changes/654b667a9b6b576b663eca96302716e09eb5270b#r2910931714)_

**Comment by chaitanya-3006** [comment]
Good point about the wording here. Calling `X` and `y` a "table" might be a bit misleading since DataOps can work with non-tabular data as well (for example images or other data structures).

Maybe it would be clearer to use a more general term like  "dataset", or "input/target data" so the docstring stays accurate regardless of the data type.

**Comment by MarieSacksick** [comment]
The options by @chaitanya-3006 seem good to me, they are consistant with other docs (and I find them personally easy to understand).  
To check what is done in other documentations when talking about images, for instance, [tensorflow](https://www.tensorflow.org/tutorials/images/classification) mention "data" or "dataset" to talk about images classification, and "input" and "target". [autogluon](https://auto.gluon.ai/0.3.0/tutorials/image_prediction/beginner.html) also use the word "dataset". They don't use words to designate input and target though.

**Comment by chaitanya-3006** [comment]
I agree with moving away from the term "table" here since DataOps can operate on non-tabular structures as well.

Using terms like:

* "dataset"
* "data"
* "input data" / "target data"

would make the docstrings more data-agnostic and consistent with how libraries like TensorFlow and AutoGluon describe image and multimodal workflows.

For example:

* `mark_as_X`: "Mark input data/features"
* `mark_as_y`: "Mark target data/labels"

or simply refer to them as "datasets" instead of "tables" throughout the docs.

**Comment by jeromedockes** [comment]
in typical use cases they will be tables and the things we mark as X and y cannot be anything because they have to be valid inputs for scikit-learn cross-validation splitters so basically dataframes or numpy arrays. so even if "tables" is a bit too restrictive I think it conveys better what they are than things like data or dataset which are too generic
